src/python/pants/core/util_rules/adhoc_binaries_test.py

```diff
@@ -36,16 +36,17 @@ def rule_runner() -> RuleRunner:
 
 @pytest.mark.parametrize("env_tgt", [None, LocalEnvironmentTarget({}, address=Address(""))])
 def test_local(env_tgt) -> None:
+    def mock_download_python_binary(
+        _: _DownloadPythonBuildStandaloneBinaryRequest,
+    ) -> _PythonBuildStandaloneBinary:
+        pytest.fail()
+
     result = run_rule_with_mocks(
         adhoc_binaries.get_python_for_scripts,
         rule_args=[EnvironmentTarget("local", env_tgt)],
-        mock_gets=[
-            MockGet(
-                output_type=_PythonBuildStandaloneBinary,
-                input_types=(_DownloadPythonBuildStandaloneBinaryRequest,),
-                mock=lambda _: pytest.fail(),
```

Comment:
Well this is weird.

Reply:
The test apparently is trying to make sure that local environment is short-circuited in the rule.